src/modules/ekf2/EKF/ekf.h

```diff
@@ -451,18 +451,27 @@ class Ekf final : public EstimatorInterface
 	void updateHorizontalDeadReckoningstatus();
 	void updateVerticalDeadReckoningStatus();
 
-	struct {
-		uint8_t velNE_counter;	///< number of horizontal position reset events (allow to wrap if count exceeds 255)
-		uint8_t velD_counter;	///< number of vertical velocity reset events (allow to wrap if count exceeds 255)
-		uint8_t posNE_counter;	///< number of horizontal position reset events (allow to wrap if count exceeds 255)
-		uint8_t posD_counter;	///< number of vertical position reset events (allow to wrap if count exceeds 255)
-		uint8_t quat_counter;	///< number of quaternion reset events (allow to wrap if count exceeds 255)
+	struct StateResetCounts
+	{
+		uint8_t velNE{0};	///< number of horizontal position reset events (allow to wrap if count exceeds 255)
+		uint8_t velD{0};	///< number of vertical velocity reset events (allow to wrap if count exceeds 255)
```

Comment:
> allow to wrap if count exceeds 255

Do we actually handle this somewhere? IIRC the logic in the controller checks if the new counter is higher than the current one so it would not apply the delta if a wrap occurs

Reply:
For the most part I think it's good enough if we check only that the reset count changed, and then we get the wrap around for free.

Reply:
Now that I think of it the estimator selector handles checking that it actually did increase by 1, and if it didn't then the actual published delta is actually the full change since previous publication. This is to handle jumping between different instances and maintaining a stable monotonically increasing reset count + deltas. Need to consider if this is good enough.

https://github.com/PX4/PX4-Autopilot/blob/c67f03f3837086b9b0032f67b71487bdf4dcad31/src/modules/ekf2/EKF2Selector.cpp#L433